### DISCUSSION TITLE
Fastflow API error

### DISCUSSION THREAD

**Discussion by harshsinha03** [open-edge-platform/anomalib#1851] [Q&A] [answered]
Hey I am new to this library and I am trying to implement Fastflow but this is the error I am getting

`model = Fastflow(
    input_size=(256, 256), 
    backbone="resnet18",
    pre_trained = True,
    flow_steps=8,
    conv3x3_only=False, 
    hidden_ratio=1.0)`

`def configure_optimizers(lightning_module: LightningModule, optimizer: Optimizer) -> Optimizer:
    del lightning_module  
    return optimizer
optimizer = Adam(params=model.parameters(), lr=0.001, betas=(0.9, 0.999), weight_decay=1e-5)
fn = partial(configure_optimizers, optimizer=optimizer)
update_wrapper(fn, configure_optimizers)  
model.configure_optimizers = MethodType(fn, model)`

`trainer = Trainer(
    accelerator="gpu",
    auto_scale_batch_size=False,
    check_val_every_n_epoch=1,
    devices=1,
    max_epochs=5,
    num_sanity_val_steps=0,
    val_check_interval=1.0,
)
trainer.fit(model=model, datamodule=datamodule)`

AttributeError: 'Fastflow' object has no attribute 'threshold_method'

**Comment by blaz-r** [comment, marked as answer]
I'm unsure if this is the only issue, but use Engine, not directly trainer from lighting. Engine is anomalib specific wrapper of trainer that does some additional things, but also takes all the params from trainer if you need those.
